services/src/main/java/org/fao/geonet/services/metadata/Insert.java

```diff
@@ -67,30 +92,72 @@ public void init(Path appPath, ServiceConfig params) throws Exception {
         this.stylePath = appPath.resolve(Geonet.Path.IMPORT_STYLESHEETS);
     }
 
-    //--------------------------------------------------------------------------
-    //---
-    //--- Service
-    //---
-    //--------------------------------------------------------------------------
-
-    public Element serviceSpecificExec(Element params, final ServiceContext context) throws Exception {
-        GeonetContext gc = (GeonetContext) context.getHandlerContext(Geonet.CONTEXT_NAME);
-
-        DataManager dataMan = gc.getBean(DataManager.class);
+    @RequestMapping(value = { "/{lang}/md.insert",
+            "/{lang}/xml.metadata.insert",
```

Comment:
Should we start dropping old URLs and try to make one API with a better doc ? Migrating services to spring could be a good opportunity to do that.